app/controlplane/pkg/data/workflowcontract.go

```diff
@@ -273,14 +277,39 @@ func (r *WorkflowContractRepo) SoftDelete(ctx context.Context, id uuid.UUID) err
 }
 
 func entContractVersionToBizContractVersion(w *ent.WorkflowContractVersion) (*biz.WorkflowContractVersion, error) {
-	contract := &schemav1.CraftingSchema{}
-	err := proto.Unmarshal(w.Body, contract)
-	if err != nil {
-		return nil, err
+	contract := &biz.Contract{
+		Raw:    w.RawBody,
+		Format: w.RawBodyFormat,
+	}
+
+	// We have two ways of storing the contract body, the old way is the body column which contains the binary representation of the proto message
+	// and the new way which is the raw_body and raw_body_format pairs
+	// Regardless of what's stored, we want to make sure we always return the contract object that contains the raw and binary representation
+	var err error
+	// Scenario 1: contracts that have been stored (and not updated) before the introduction of the raw_body field will have an empty raw_body
+	// so we will generate a json representation of the contract to populate the raw_body field in that case
+	// that way clients can always expect a raw_body field to be present
+	if len(contract.Raw) == 0 {
```

Comment:
I know it's the same, but I would put here `len(w.RawBody)`, just to keep consistency with the `else if` below.